python/cudf/cudf/utils/applyutils.py

```diff
@@ -111,12 +111,27 @@ def apply_chunks(
 
 
 def make_aggregate_nullmask(df, columns=None, op="and"):
+    from cudf import MultiIndex
+
     out_mask = None
-    for k in columns or df.columns:
-        if not df[k].nullable:
-            continue
+    for k in columns or df._data:
+        try:
+            if not df[k].nullable:
+                continue
+            nullmask = df[k].nullmask
+        except KeyError:
```

Comment:
I typically prefer to check if something exists as opposed to just catching the exception. But would be curious for others thoughts as well.

Reply:
I tried an example with `try and except` and with `search`, overall `try and except` looks better compared to other option
```
In [46]: import cudf                                                                                                                                                                                                             

In [47]: df = cudf.DataFrame({"a":[1], "b":[2], "c":[1], "d":[3], "e":[4]})                                                                                                                                                      

In [48]: df = df.set_index('a')                                                                                                                                                                                                  

In [49]: def sear(df, col): 
    ...:     if col in df.index._data: 
    ...:         return df.index._data[col] 
    ...:     else: 
    ...:         return df._data[col] 
    ...:                                                                                                                                                                                                                         

In [50]: def try_ex(df, col): 
    ...:     try: 
    ...:         return df._data[col] 
    ...:     except KeyError: 
    ...:         return df.index._data[col] 
    ...:          
    ...:                                                                                                                                                                                                                         

In [51]: %timeit try_ex(df, 'a')                                                                                                                                                                                                 
895 ns ± 0.569 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [52]: %timeit sear(df, 'a')                                                                                                                                                                                                   
868 ns ± 1.14 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [53]: %timeit try_ex(df, 'e')                                                                                                                                                                                                 
307 ns ± 0.431 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [54]: %timeit sear(df, 'e')                                                                                                                                                                                                   
971 ns ± 0.694 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

```

Reply:
You have the logic backwards 😄 It searches in `df._data` before searching in `df.index._data`.

Reply:
For try_except I left it after as I had to take care of situation `index` keyword. And didn't want to search again in `df._data` for it.

Reply:
With switched logic 
```
In [56]: def sear(df, col): 
    ...:     if col in df._data: 
    ...:         return df._data[col] 
    ...:     else: 
    ...:         return df.index._data[col] 
    ...:          
    ...:                                                                                                                                                                                                                         

In [57]: %timeit sear(df, 'a')                                                                                                                                                                                                   
975 ns ± 0.448 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [58]: %timeit sear(df, 'e')                                                                                                                                                                                                   
641 ns ± 1.16 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

```

Reply:
Seems like try except is more performant, lets go with that for now 😄